runtime/compiler/trj9/control/rossa.cpp

```diff
@@ -1136,32 +1136,39 @@ onLoadInternal(
          jitConfig->codeCacheTotalKB = 64 * 1024;
          jitConfig->dataCacheTotalKB = 192 * 1024;
 #endif
+
 #if defined(J9ZTPF)
-#define ZTPF_PERCENTAGE_OF_MEMORY 20;
          /*
           * The z/TPF OS does not have the ability to reserve memory. It allocates whatever it
           * is asked to allocate. Allocate the code cache based on a percentage (20%) of memory
-          * the process is allowed to have; cap the maximum at 256m and use a minimum of 1m for
-          * a small MAXXMMES. It is found in the z/TPF cinfc table entry CMMMMES.
+          * the process is allowed to have; cap the maximum using the default 256m and use a 1m floor.
           * MAXXMMES is specified as the number of 1MB frames.
-          * Create a 2048 KB data cache. This set of default values for the z/TPF os can be
+          * Use the default 2048 KB data cache. The default values for z/TPF can be
           * overriden via the command line interface.
           */
-         uint16_t physMemory = *(uint16_t*)(cinfc_fast(CINFC_CMMMMES) + 8);
-         if (physMemory < 64)
+
+#define ZTPF_CODE_CACHE_DIVISOR 5
```

Comment:
Would it be possible to use a `static const uint16_t` here instead for type safety? Perhaps documenting what this is doing exactly could also be worth while?

Reply:
I got a compilation error changing this to const.  I was able to update this to static.

We are making some more mods in here very shortly and some of this code is going away for the 2Q refresh.  We just wanted to make sure we got 20% of our total memory available for the 1Q refresh instead of 5%.

Reply:
Sorry to be the jerk here, @jjohnst-us .

This project does not pursue short term "release" goals for downstream consumers. The project can choose to make short-term decisions for OpenJ9 releases for platforms that are supported by Eclipse OpenJ9, but unless we can arrange to have z/TPF machines to build and test in the open, I'm afraid you're not likely to ever fit into that category. Even so, I would prefer those kinds of changes to be made in a release branch rather than introducing churn onto our master branch.

If you need to make a change for your "1Q refresh" then I would kindly suggest you make those changes in your local copy of the code and use this repository for the more strategic implementation for your platform releases. Otherwise, you're just adding churn to an already busy set of committers on this project that does not (technically) benefit this open project. Yours will not be the first downstream consumer to manage these kinds of changes downstream from OpenJ9, and as you said yourself, their lifetime is limited.

Reply:
+1 to what @mstoodle said. I'd like to add, as someone who loves performance, that if you have data/reasoned argument to say this change is a good change to make for performance reasons in general for the long term then we can certainly review/debate the change on the merits of that data/argument.

Reply:
Not sure what you mean by 'short term' release goals.  In our lab this is considered a defect, as our original design/discussion by our architects was to use 20% not 5%.  This affects only our platform and that is why its in platform specific guard macros.

Isn't having a seperate release branch defeat the purpose of single source?

Should we have one z/TPF committer representative of the z/TPF platform to alleviate the 'busy' set of commiters?  At least I have seen that in many other "open" projects.

Reply:
@mstoodle Also consider we have introduce one or two build breaks throughout our entire port.

Reply:
@jjonst-us: Possibly I misinterpreted your earlier statements. By 'short term' release goals I was keying off your statement "We are making some more mods in here very shortly and some of this code is going away for the 2Q refresh"

That sounds to me like you're under pressure to commit things to this project for *your* Q1 release where you know or strongly suspect those things will later be undone for *your* Q2 release. 

Did I misinterpret your statement?